classes/form/settings_form.php

```diff
@@ -111,66 +110,24 @@ public function define_file_transfer_section($mform, $config) {
         return $mform;
     }
 
-    public function define_amazon_s3_check($mform, $config) {
-        global $OUTPUT;
-        $connection = false;
-
-        $client = new s3_client($config);
-        $connection = $client->test_connection();
+    public function define_client_section($mform, $config) {
+        global $CFG, $OUTPUT;
 
-        if ($connection->success) {
-            $mform->addElement('html', $OUTPUT->notification($connection->message, 'notifysuccess'));
+        if (empty($CFG->alternative_file_system_class)) {
```

Comment:
We're lose the ability to pre-set client config before we flip on the alternative_file_system_class - do you think this is important enough to change how we handle this ?

Couple alternatives i can think of:

1. Allow client to set which object storage they want to use as a setting. We can then also check that alternative_file_system_class matches the expected class for that client and warn if it doesnt.
2. Just show all client sections here if alternative_file_system_class isn't set with appropriate text that alternative_file_system_class isnt to their respective classes in each section